### PR TITLE
Add aditional logged items

### PR DESCRIPTION
**Pull request by noamzbr**
Add confusion matrix and train metrics, attempt to log model object.

### CODE REVIEW

File: soundbay/utils/logging.py [hunk 1]
```diff
@@ -105,24 +104,27 @@ def update_predictions(self, pred_tuple):
         self.pred_proba_list.append(torch.softmax(pred_tuple[0].data, 1).cpu().numpy())
         self.label_list += label.cpu().numpy().tolist()
 
-    def calc_metrics(self, epoch):
+    def calc_metrics(self, epoch: int, mode: str = 'train', label_names: List[str] = ('Noise', 'Call')):
```
Comment (mosheman5) [inline soundbay/utils/logging.py:107]: thank you :)

**Review by mosheman5 — DISMISSED** [review]
lgtm

File: soundbay/utils/logging.py [hunk 2]
```diff
@@ -105,24 +104,27 @@ def update_predictions(self, pred_tuple):
...
-        self.log_writer.log({'f1score': self.metrics_dict['f1score']}, step=epoch)
-        self.log_writer.log({'precision': self.metrics_dict['precision']}, step=epoch)
-        self.log_writer.log({'recall': self.metrics_dict['recall']}, step=epoch)
-        if not self.debug_mode:
-            self.log_writer.log({'auc': self.metrics_dict['auc']}, step=epoch)
+        self.log_writer.log({f'{mode}_Accuracy': self.metrics_dict['accuracy']}, step=epoch)
+        self.log_writer.log({f'{mode}_f1score': self.metrics_dict['f1score']}, step=epoch)
+        self.log_writer.log({f'{mode}_precision': self.metrics_dict['precision']}, step=epoch)
+        self.log_writer.log({f'{mode}_recall': self.metrics_dict['recall']}, step=epoch)
+        if not self.debug_mode and mode != 'train':
```
Comment (mosheman5) [inline soundbay/utils/logging.py:115]: actually, why not log confusion matrix or pr curve on training as well? Might be an important indicator on the sources of bugs (overfit vs split distribution)

**Comment by tomer-grin** [comment]
Cool changes, useful metrics!
Nothing further to add.

**Pull request merged**